Models

barbarakos opened the pull request:
Added models for every layer.

barbarakos committed:
Models

Added models for every layer.

dorastipkovic commented on QuizApp/User/ViewModel/UserViewModel.swift:
```diff
@@ -19,7 +19,7 @@ class UserViewModel {
         Task {
             do {
-                let user: UserResponseModel = try await useCase.getUser()
+                let user: UserModel = UserModel(from: try await useCase.getUser())
```
```suggestion
                let user = UserModel(from: try await useCase.getUser())
```

dorastipkovic commented on QuizApp/User/UseCase/UserUseCase.swift:
```diff
@@ -18,12 +18,13 @@ class UserUseCase: UserUseCaseProtocol {
         }
 
-        return try await dataSource.getUser(accessToken: accessToken)
+        let user = UserUseCaseModel(from: try await dataSource.getUser(accessToken: accessToken))
+        return user
```
```suggestion
        return UserUseCaseModel(from: try await dataSource.getUser(accessToken: accessToken))
```

dorastipkovic commented on QuizApp/Quiz/Models/QuizResponseModel.swift:
```diff
@@ -0,0 +1,13 @@
+
+    let category: String
+    let description: String
+    let difficulty: String
```
Difficulty has expected values, so it should be a enum. 
You need to create models for each layer, just like you did now with oteher models. 
e.g. DifficultyResponseModel, DifficultyDataSourceModel etc

dorastipkovic commented:
If a model has property `id`, move it to top. 
Rename inner arguments name in model initializers so it looks like this: 

```
init(from model: SomeWhateverModel) {
}
```

dorastipkovic commented:
Move model initializers to extensions.

barbarakos committed:
Requested changes

Added:
- DifficultyResponseModel
- DifficultyDataModel
- DifficultyUseCaseModel
- DifficultyModel

Moved model initializers to extensions, renamed inner arguments name in model initializers.
